File: src/main/java/com/baulsupp/oksocial/completion/CompletionQuery.java
package com.baulsupp.oksocial.completion;

import com.baulsupp.oksocial.authenticator.AuthUtil;
import java.util.List;
import java.util.Map;
import java.util.concurrent.CompletableFuture;
import okhttp3.OkHttpClient;
import okhttp3.Request;

import static java.util.stream.Collectors.toList;

public class CompletionQuery {
  public static CompletableFuture<List<String>> getIds(OkHttpClient client, String urlString,
      String path,
      String key) {
    Request request = new Request.Builder().url(urlString).build();

    return AuthUtil.enqueueJsonMapRequest(client, request)
        .thenApply(map -> {
          List<Map<String, Object>> surveys = (List<Map<String, Object>>) map.get(path);
          return surveys.stream().map(m -> (String) m.get(key)).collect(toList());
        });
  }
}

File: src/main/java/com/baulsupp/oksocial/completion/CompletionQuery.kt
package com.baulsupp.oksocial.completion

import com.baulsupp.oksocial.authenticator.AuthUtil
import java.util.concurrent.CompletableFuture
import okhttp3.OkHttpClient
import okhttp3.Request

import java.util.stream.Collectors.toList

object CompletionQuery {
    fun getIds(client: OkHttpClient, urlString: String,
               path: String,
               key: String): CompletableFuture<List<String>> {
        val request = Request.Builder().url(urlString).build()

        return AuthUtil.enqueueJsonMapRequest(client, request)
                .thenApply { map ->
                    val surveys = map[path] as List<Map<String, Any>>
                    surveys.stream().map { m -> m[key] as String }.collect<List<String>, Any>(toList())
                }
    }
}
